packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -144,6 +146,50 @@ def test_if_we_have_capabilities(cluster):
     assert {'name': 'PACKAGE_MANAGEMENT'} in r.json()['capabilities']
 
 
+def test_cosmos_package_add(cluster):
+    def package_add_configured():
+        user_config = load_yaml("/opt/mesosphere/etc/user.config.yaml")
+
+        is_staged_uri_set = user_config.get(
+            'cosmos_config',
+            {}
+        ).get('staged_package_storage_uri')
+
+        is_package_uri_set = user_config.get(
+            'cosmos_config',
+            {}
+        ).get('staged_package_storage_uri')
+
+        return is_staged_uri_set and is_package_uri_set
+
+    r = cluster.post(
+        '/package/add',
+        headers={
+            'Accept': (
+                'application/vnd.dcos.package.add-response+json;'
+                'charset=utf-8;version=v1'
+            ),
+            'Content-Type': (
+                'application/vnd.dcos.package.add-request+json;'
+                'charset=utf-8;version=v1'
+            )
+        },
+        json={
+            'packageName': 'cassandra',
+            'packageVersion': '1.0.20-3.0.10'
+        }
+    )
+
+    if package_add_configured():
+        # if the config is enabled then Cosmos should accept the request and
+        # return 202
+        assert r.status_code == 202
```

Comment:
`assert r.status_code == 202` ends up producing really hard to debug / track down errors when it fails (you don't get the status code or response body printed which are both usually critical in solving). Please use r.raise_for_status() or do a `assert r.status_code == 202, "status_code: {}, content: {}".format(r.status_code, r.content))`